coderd/users_test.go

```diff
@@ -704,15 +704,48 @@ func TestUpdateUserProfile(t *testing.T) {
 		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
 		defer cancel()
 
-		_, _ = client.User(ctx, codersdk.Me)
+		me, err := client.User(ctx, codersdk.Me)
+		require.NoError(t, err)
+
 		userProfile, err := client.UpdateUserProfile(ctx, codersdk.Me, codersdk.UpdateUserProfileRequest{
-			Username: "newusername",
-			Name:     "Mr User",
+			Username: me.Username + "1",
+			Name:     me.Name + "1",
 		})
+		numLogs++ // add an audit log for user update
+
 		require.NoError(t, err)
-		require.Equal(t, userProfile.Username, "newusername")
-		require.Equal(t, userProfile.Name, "Mr User")
+		require.Equal(t, me.Username+"1", userProfile.Username)
+		require.Equal(t, me.Name+"1", userProfile.Name)
+
+		require.Len(t, auditor.AuditLogs(), numLogs)
+		require.Equal(t, database.AuditActionWrite, auditor.AuditLogs()[numLogs-1].Action)
+	})
+
+	t.Run("UpdateSelfAsMember", func(t *testing.T) {
+		t.Parallel()
+		auditor := audit.NewMock()
+		client := coderdtest.New(t, &coderdtest.Options{Auditor: auditor})
+		numLogs := len(auditor.AuditLogs())
```

Comment:
Why do we need these?

Reply:
We want to assert that the change gets audit-logged.